panic: runtime error: index out of range

If I try to use the latest version of go-bindata within https://github.com/go-gitea/gitea i run into the below error.

I try to execute ```go-bindata -o=modules/bindata/bindata.go -ignore="\\.DS_Store|README" -pkg=bindata conf/...```. Another contributor of gitea built https://github.com/jteeuwen/go-bindata/commit/81e20ed1b787aee59470382444baf4cb6f9ae6f0 manually and it worked as before. Is there any proper fix?

```
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/jteeuwen/go-bindata.safeFunctionName(0xc20801e360, 0x12, 0xc20803c450, 0x0, 0x0)
	/home/tboerger/Projects/golang/src/github.com/jteeuwen/go-bindata/convert.go:235 +0x955
github.com/jteeuwen/go-bindata.findFiles(0xc20802a690, 0xe, 0x0, 0x0, 0x1, 0xc20801e100, 0xc20803a038, 0x1, 0x1, 0xc20803c450, ...)
	/home/tboerger/Projects/golang/src/github.com/jteeuwen/go-bindata/convert.go:210 +0xca9
github.com/jteeuwen/go-bindata.findFiles(0x7fff2667600a, 0x4, 0x0, 0x0, 0x1, 0xc20801e100, 0xc20803a038, 0x1, 0x1, 0xc20803c450, ...)
	/home/tboerger/Projects/golang/src/github.com/jteeuwen/go-bindata/convert.go:176 +0x690
github.com/jteeuwen/go-bindata.Translate(0xc20805a000, 0x0, 0x0)
	/home/tboerger/Projects/golang/src/github.com/jteeuwen/go-bindata/convert.go:34 +0x27e
main.main()
	/home/tboerger/Projects/golang/src/github.com/jteeuwen/go-bindata/go-bindata/main.go:20 +0x45

goroutine 2 [runnable]:
runtime.forcegchelper()
	/usr/lib64/go/src/runtime/proc.go:90
runtime.goexit()
	/usr/lib64/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 3 [runnable]:
runtime.bgsweep()
	/usr/lib64/go/src/runtime/mgc0.go:82
runtime.goexit()
	/usr/lib64/go/src/runtime/asm_amd64.s:2232 +0x1

goroutine 4 [runnable]:
runtime.runfinq()
	/usr/lib64/go/src/runtime/malloc.go:712
runtime.goexit()
	/usr/lib64/go/src/runtime/asm_amd64.s:2232 +0x1
```

--- a/convert.go
+++ b/convert.go
@@ -223,17 +223,17 @@ var regFuncName = regexp.MustCompile(`[^a-zA-Z0-9_]`)
 // prevent conflict based on name translation.
 func safeFunctionName(name string, knownFuncs map[string]int) string {
 	var inBytes, outBytes []byte
+	var toUpper bool
 
 	name = strings.ToLower(name)
 	inBytes = []byte(name)
 
 	for i := 0; i < len(inBytes); i++ {
 		if regFuncName.Match([]byte{inBytes[i]}) {
-			for j := i; j < len(inBytes) && regFuncName.Match([]byte{inBytes[j]}); j++ {
-				i++
-			}
+			toUpper = true
+		} else if toUpper {
 			outBytes = append(outBytes, []byte(strings.ToUpper(string(inBytes[i])))...)
-			// bytes[i] = strings.ToUpper(string(byte))
+			toUpper = false
 		} else {
 			outBytes = append(outBytes, inBytes[i])
 		}

